android/src/main/java/com/reactlibrary/googlesignin/RNGoogleSignInModule.java

```
@@ -186,7 +186,12 @@ public void onNewIntent(Intent intent) {
     public void onActivityResult(Activity activity, final int requestCode, final int resultCode, final Intent intent) {
         if (requestCode == RC_SIGN_IN) {
             GoogleSignInResult result = Auth.GoogleSignInApi.getSignInResultFromIntent(intent);
-            handleSignInResult(result);
+            // A workaround to prevent a crash and the bug of google sign in api which gives an error for the second attempt to sign in
+            if (result == null) {
+                signIn();
```

I don't think we should do this here. I think there are cases where sign in can result in null for certain users, and this could potentially cause an infinite loop. We should probably just throw a react error if the result is null for any reason.

---

We can move the null handling to the `handleSignInResult` method as well.